README.md

```diff
@@ -473,22 +485,193 @@ When the scanner successfully decodes a `TransactionRequest` it will call its de
+* `onTransactionConsumptionFinalizedSuccess(TransactionConsumption)`: Invoked if a `TransactionConsumption` has been finalized successfully, and the transfer was made between the 2 addresses.
+* `onTransactionConsumptionFinalizedFail(APIError)`: Invoked if a `TransactionConsumption` fails to consume the request.
+        
+### TransactionConsumption Event
+ 
+Similarly to the `TransactionRequestCallback`, a `TransactionConsumption` can be listened for incoming confirmations using the `TransactionConsumptionCallback`.
+The possible events are:
+* `onTransactionConsumptionFinalizedSuccess(TransactionConsumption)`: Invoked if a `TransactionConsumption` has been finalized successfully, and the transfer was made between the 2 addresses.
+* `onTransactionConsumptionFinalizedFail(APIError)`: Invoked if a `TransactionConsumption` fails to consume the request.
+
+**Usage**
+```kotlin
+// The transaction requestor listen for the event 
+transactionRequest.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionRequestCallback() {
+   override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+       // Do something
+   }
+
+   override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+       // Do something
+   }
+})
+
+// The transaction consumer listen for the event
+transactionConsumption.startListeningEvents(socketClient, object: SocketCustomEventCallback.TransactionConsumptionCallback() {
+  override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
+      // Do something
+  }
+})
+```
+
+> Note: You might want to listen for the event later after you got the `TransactionRequest` or `TransactionConsumption` object, but you might not want to pass the object around.
+In this case, it is possible to keep only the `socket_topic` string directly. Then you can alternatively listening for custom events by using `joinChannel` method of the `OMGSocketClient` instance.
+This implementation will provide the exactly same result as the implementation above ☝️. For example,
+
+```kotlin
+// The transaction requestor listen for the event 
+socketClient.joinChannel(SocketTopic(transactionRequestTopic), listener = object: SocketCustomEventCallback.TransactionRequestCallback(){
+  override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+      // Do something
+  }
+
+  override fun onTransactionConsumptionFinalizedFail(apiError: APIError) {
```

Comment:
Why not pass the `TransactionConsumption` context on this one? Might be useful to properly handle the error no? 🤔

Reply:
Yes, it should be nice. Added it 👍